src/dispatch/database/service.py

```diff
@@ -558,7 +558,8 @@ def search_filter_sort_paginate(
 
         tag_all_filters = []
         if filter_spec:
-            filter_spec = json.loads(filter_spec)
+            if isinstance(filter_spec, str):
+                filter_spec = json.loads(filter_spec)
             query = apply_filter_specific_joins(model_cls, filter_spec, query)
```

Comment:
nit: it'd be worth adding a comment to explain the when/why/how for these conditions: `filter_spec` can be `None`, `dict`, or `str`.

the typo annotation should also be updated